modules/dnn/src/layers/fully_connected_layer.cpp

```diff
@@ -116,7 +116,6 @@ class FullyConnectedLayerImpl CV_FINAL : public InnerProductLayer
             CV_CheckEQ(inputs.size(), (size_t)2, "");
             numOutput = inputs[1].back();
             cAxis = inputs[0].size() - 1;
-            CV_CheckEQ(numOutput, inputs[0][cAxis - 1], "");
```

Comment:
Please add some test (for reference: https://github.com/opencv/opencv_extra/pull/755 + https://github.com/opencv/opencv/pull/17233)

Reply:
> Please add some test (for reference: [opencv/opencv_extra#755](https://github.com/opencv/opencv_extra/pull/755) + #17233)

Actually, this check makes no sense to me, MatMul operation doesn't require this condition. This check actually does not allow multiplying matrices with dimensions like 12x500 and 500x1000